auction-cordapp/workflows/src/main/java/net/corda/samples/flows/CreateAuctionFlow.java

```diff
@@ -0,0 +1,118 @@
+import net.corda.core.contracts.UniqueIdentifier;
+import net.corda.core.flows.*;
+import net.corda.core.identity.Party;
+import net.corda.core.transactions.SignedTransaction;
+import net.corda.core.transactions.TransactionBuilder;
+import net.corda.core.utilities.ProgressTracker;
+import net.corda.samples.contracts.AuctionContract;
+import net.corda.samples.states.AuctionState;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+/**
+ * This flow is used to create a auction for an asset.
+ */
+public class CreateAuctionFlow {
+
+    private CreateAuctionFlow(){}
+
+    @InitiatingFlow
+    @StartableByRPC
+    public static class Initiator extends FlowLogic<SignedTransaction> {
+
+        private final ProgressTracker progressTracker = new ProgressTracker();
+
+        private final Amount<Currency> basePrice;
+        private final UUID auctionItem;
+        private final LocalDateTime bidDeadLine;
+
+        /**
+         * Constructor to initialise flow parameters received from rpc.
+         *
+         * @param basePrice of the asset to be put of auction.
+         * @param auctionItem is the uuid of the asset to be put on auction
+         * @param bidDeadLine is the time till when the auction will be active
+         */
+        public Initiator(Amount<Currency> basePrice, UUID auctionItem, LocalDateTime bidDeadLine) {
+            this.basePrice = basePrice;
+            this.auctionItem = auctionItem;
+            this.bidDeadLine = bidDeadLine;
+        }
+
+        @Override
+        public ProgressTracker getProgressTracker() {
+            return progressTracker;
+        }
+
+        @Suspendable
+        @Override
+        public SignedTransaction call() throws FlowException {
+            // Fetch the notary for the transaction
+            Party notary = getServiceHub().getNetworkMapCache().getNotaryIdentities().get(0);
+            Party auctioneer = getOurIdentity();
+
+            // Fetch all parties from the network map and remove the auctioneer and notary. All the parties are added as
+            // participants to the auction state so that its visible to all the parties in the network.
+            List<Party> bidders = getServiceHub().getNetworkMapCache().getAllNodes().stream()
+                    .map(nodeInfo -> nodeInfo.getLegalIdentities().get(0))
+                    .collect(Collectors.toList());
+            bidders.remove(auctioneer);
+            bidders.remove(notary);
+
+            // Create the output state. Use a linear pointer to point to the asset on auction. The asset would be added
+            // as a reference state to the transaction and hence we won't spend it.
+            AuctionState auctionState = new AuctionState(
+                    new LinearPointer<>(new UniqueIdentifier(null, auctionItem), LinearState.class),
```

Comment:
Could also do a toPointer() from the Asset class. I don't think it's important but could also show that having helpers in the state can be useful as it's done in some other cases such as bundling the linear pointer and tokenType in TokenPointer in TokenSDK (although this is much simpler...)

Reply:
Hmmm.. yes fair point, but to use a toPointer() I would first have to create an instance of the Asset, which isn't needed in the flow. Kind-of an over-kill, I guess.